.gitlab/pipeline_on_merge.yml

```diff
@@ -1,20 +1,43 @@
-# Copyright (c) 2024 ETH Zurich
-#
-# SPDX-License-Identifier: BSL-1.0
-# Distributed under the Boost Software License, Version 1.0. (See accompanying
-# file BOOST_LICENSE_1_0.rst or copy at http://www.boost.org/LICENSE_1_0.txt)
-
 include:
+  - local: '.gitlab/includes/skip_job.yml'
```

Comment:
An alternative would perhaps be to put all jobs back in `pipeline.yml`, and skip the same set of jobs there instead. Then we wouldn't need the `ci-on-merge` pipeline at all. As far as I can tell this would be equivalent, but simpler as there's only one pipeline and there would be no need for the dummy job.